src/main/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -1154,12 +1168,35 @@ public Object getAttribute(final String tag) {
         return getAttribute(SAMTag.makeBinaryTag(tag));
     }
 
+    /**
+     * Get the value for a SAM tag.
+     * WARNING: Some value types (e.g. byte[]) are mutable.  It is dangerous to change one of these values in
+     * place, because some SAMRecord implementations keep track of when attributes have been changed.  If you
+     * want to change an attribute value, call setAttribute() to replace the value.
+     *
+     * @param tag Two-character tag name.
+     * @return Appropriately typed tag value, or null if the requested tag is not present.
+     */
+    public Object getAttribute(final SAMTag tag) {
+        return getAttribute(tag.getBinaryTag());
+    }
+
     /**
      * Get the tag value and attempt to coerce it into the requested type.
      * @param tag The requested tag.
      * @return The value of a tag, converted into a signed Integer if possible.
      * @throws RuntimeException If the value is not an integer type, or will not fit in a signed Integer.
      */
+    public Integer getIntegerAttribute(final SAMTag tag) {
+        return getIntegerAttribute(tag.name());
```

Comment:
We *could* introduce overloads of all these methods that take `short` intead to avoid some unnecessary conversion.  I don't know if it's worth it though.

Reply:
well….that isn't what _this_ PR is about…. :-)